runtime/tests/checker/path_test.go

```diff
@@ -105,3 +105,44 @@ func TestCheckPath(t *testing.T) {
 		assert.IsType(t, &sema.InvalidPathDomainError{}, errs[0])
 	})
 }
+
+func TestConvertStringToPath(t *testing.T) {
+	t.Parallel()
+
+	domainTypes := map[common.PathDomain]sema.Type{
+		common.PathDomainStorage: sema.StoragePathType,
+		common.PathDomainPublic:  sema.PublicPathType,
+		common.PathDomainPrivate: sema.PrivatePathType,
+	}
+
+	test := func(domain common.PathDomain) {
+
+		t.Run(fmt.Sprintf("valid: %s", domain.Identifier()), func(t *testing.T) {
+
+			t.Parallel()
+
+			domainType := domainTypes[domain]
+
+			checker, err := ParseAndCheck(t,
+				fmt.Sprintf(
+					`
+                      let x: %[1]s? = %[1]s("/%[2]s/foo")
+                    `,
+					domainType.String(),
+					domain.Identifier(),
```

Comment:
Shouldn't the call require an argument label `identifier:`? Maybe add a test that checks the argument label is required.

Also, shouldn't the call only accept an identifier and not a full path? (Is not checked statically, but still)

Maybe also remove the type annotation for the variable declaration: As it is explicitly given, the assertion below always succeeds. Instead, you want to assert that the inferred type for the variable matches the return type of the converter function:

```suggestion
                      let x = %s(identifier: "foo")
                    `,
					domainType.String(),
```